docs/reference/config.md

```diff
@@ -265,20 +295,38 @@ The following settings are available:
   :::
 : The AWS KMS key Id to be used to encrypt files stored in the target S3 bucket.
 
+`aws.client.targetThroughputInGbps`
+: :::{versionadded} 25.06.0-edge
+  :::
+: The S3 Async client target network throughput in Gbps. This value is used to automatically set `maxConcurrency` and `maxNativeMemory`. (Default: `10`). See [targetThroughputInGbps](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3CrtAsyncClientBuilder.html#targetThroughputInGbps(java.lang.Double)) for more details.
+
+`aws.client.multipartThreshold`
+: :::{versionadded} 25.06.0-edge
+  :::
+: The S3 Async client threshold to create multipart S3 transfers. Default is the same as `minimumPartSize` https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3CrtAsyncClientBuilder.html#thresholdInBytes(java.lang.Long)
+
+`aws.client.tmThreads`
+: :::{versionadded} 25.06.0-edge
+  :::
+: Number of threads used by the S3 transfer manager. (default `10`).
+
 `aws.client.userAgent`
+:::{deprecated} 25.06.0-edge
+  This option is no longer supported.
+  :::
 : The HTTP user agent header passed with all HTTP requests.
 
 `aws.client.uploadChunkSize`
-: The size of a single part in a multipart upload (default: `100 MB`).
+: The size of a single part in a multipart upload (default: `100 MB`). Since version 25.06.0-edge, this option only applies to uploads generated from OutputStream. Other uploads are managed by the S3 transfer manager.
```

Comment:
Is it worth keeping these options if they are only applied to OutputStream's ? We try not to expose this interface directly to users, in favor of `Path`, so I would be inclined to remove these options.